pkg/server/api/trustdomain/v1/service.go

```diff
@@ -64,3 +80,75 @@ func (s *Service) BatchDeleteFederationRelationship(ctx context.Context, req *tr
 func (s *Service) RefreshBundle(ctx context.Context, req *trustdomainv1.RefreshBundleRequest) (*emptypb.Empty, error) {
 	return nil, status.Error(codes.Unimplemented, "unimplemented")
 }
+
+func fieldsFromRelationshipProto(proto *types.FederationRelationship, mask *types.FederationRelationshipMask) logrus.Fields {
+	fields := logrus.Fields{}
+
+	if mask == nil {
+		mask = protoutil.AllTrueFederationRelationshipMask
+	}
+
+	if proto == nil {
+		return fields
+	}
+
+	if proto.TrustDomain != "" {
+		fields[telemetry.TrustDomainID] = proto.TrustDomain
+	}
+
+	if mask.BundleEndpointUrl {
+		fields[telemetry.BundleEndpointURL] = proto.BundleEndpointUrl
+	}
+
+	if mask.BundleEndpointProfile {
+		switch profile := proto.BundleEndpointProfile.(type) {
+		case *types.FederationRelationship_HttpsWeb:
+			fields[telemetry.BundleEndpointProfile] = datastore.BundleEndpointWeb
+		case *types.FederationRelationship_HttpsSpiffe:
+			// TODO: may we set bundle as field? profile.HttpsSpiffe.Bundle
+			fields[telemetry.BundleEndpointProfile] = datastore.BundleEndpointSPIFFE
+			fields[telemetry.EndpointSpiffeID] = profile.HttpsSpiffe.EndpointSpiffeId
+
+			bundleFields := api.FieldsFromBundleProto(proto.GetHttpsSpiffe().Bundle, nil)
+			for key, value := range bundleFields {
+				fields[key] = value
+			}
+		}
+	}
+
+	return fields
+}
+
+func (s *Service) createFederationRelationship(ctx context.Context, f *types.FederationRelationship, outputMask *types.FederationRelationshipMask) *trustdomainv1.BatchCreateFederationRelationshipResponse_Result {
+	log := rpccontext.Logger(ctx)
+
+	dsFederationRelationship, err := api.ProtoToFederationRelationship(f)
+	if err != nil {
+		return &trustdomainv1.BatchCreateFederationRelationshipResponse_Result{
+			Status: api.MakeStatus(log, codes.InvalidArgument, "failed to convert federation relationship", err),
+		}
+	}
```

Comment:
We need to make reject requests to create a federation relationship with our own trust domain.